metaci/db/management/commands/populate_db.py

```diff
@@ -1,59 +1,71 @@
 from django.db import transaction
-
 from guardian.shortcuts import assign_perm
 
-import factory.random
-
 from metaci import conftest as fact
-from metaci.users.models import User
-from metaci.testresults.models import TestResult
-from django.contrib.auth.models import Group
-
 from metaci.build.models import BuildFlow
-
-import random
+from metaci.cumulusci.models import Org
+from metaci.plan.models import Plan, PlanRepository
+from metaci.repository.models import Repository
+from metaci.testresults.models import TestResult
+from metaci.users.models import User
 
 
 class Command(BaseCommand):
     help = "Populates a testing DB with sample data"
 
-    def handle(self, *args, **options):
-        if self.check_non_destructive():
-            print("Populating test data")
-            # reset_db.Command().handle(*args, **options)
-            # migrate.Command().handle(*args, **options)
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--results", action="store_true", help="Create fake builds and test results"
+        )
 
-            self.create_objs()
-        else:
-            print(
+    def handle(self, *args, **options):
+        if options["results"]:
+            if self.check_non_destructive():
+                print("Populating test data")
+                # reset_db.Command().handle(*args, **options)
+                # migrate.Command().handle(*args, **options)
+
+                self.create_builds_with_test_results()
+            else:
+                print(
+                    """
+                This does not seem to be an empty or test database!
+                Please clear the database before continuing!
+                One way of clearing the database is:
+
+                python manage.py reset_db -c && python manage.py migrate && python manage.py createsuperuser --username clark --email kent@krypton.com
+
+                You can add a --noinput to that if you're sure you know what you're doing.
                 """
-            This does not seem to be an empty or test database!
-            Please clear the database before continuing!
-            One way of clearing the database is:
-
-            python manage.py reset_db -c && python manage.py migrate && python manage.py createsuperuser --username clark --email kent@krypton.com
-
-            You can add a --noinput to that if you're sure you know what you're doing.
-            """
-            )
+                )
+        else:
+            self.create_test_repo()
 
     def check_non_destructive(self):
         return User.objects.count() < 3 and TestResult.objects.count() < 10
 
-    def create_objs(self):
+    def create_builds_with_test_results(self):
         with transaction.atomic():
             factory.random.reseed_random("TOtaLLY RaNdOM")
             random.seed("RaNDOM!! TOtaLLY")
             fact.UserFactory()
             fact.UserFactory()
+            existing_plans = Plan.objects.all()
```

Comment:
Why do you rely on plans existing instead of creating them? If this is a fresh database where would they come from? Have I missed something?

Reply:
@prescod there are some default plans created in one of the database migration steps. The PlanRepositoryFactory was relying on those before too, I just moved that here so that the PlanRepositoryFactory is also useful in situations where there are no existing plans, such as a test case.